### DISCUSSION TITLE
Pinterest: merging data returned from promise.all

### DISCUSSION THREAD

**Discussion by katherinevfry** [nss-evening-cohort-14/evening-client-side#185] [Help Me!] [answered]
Katy Fry
Have you searched the other help tickets?
Yep!

Descript your issue:
I'm calling a promise.all that resolves user info and public pins. I need to write a function that checks to see if the uid of the pin matches the uid of a user, and then appends the user info onto each pin object if the user uid matches. 

List any error messages that appear:
N/A

What have you googled? 
https://scotch.io/glossary/javascript/array-find-method
https://www.javascripttutorial.net/object/javascript-merge-objects/
https://stackoverflow.com/questions/7364150/find-object-by-id-in-an-array-of-javascript-objects

What have you tried? Tell us everything you have tried.
This is what I'm currently trying:
```
mergeUserPins().then((userPinsObject) => {
      const pubPins = userPinsObject.publicPins;
      const userInfo = userPinsObject.user;
      const combinedData = pubPins.filter(() => uid === userInfo.uid);
      console.warn(combinedData);
    });
```
All I'm getting back is an empty array in the console. I'm not even sure if I'm on the right track at this point. Ha.

Here's a link to the code
https://github.com/katherinevfry/ASSIGNMENT-Firebase-Pinterest/blob/merge-user-pins/src/javascripts/components/events/navEvents.js#L18
https://github.com/katherinevfry/ASSIGNMENT-Firebase-Pinterest/blob/merge-user-pins/src/javascripts/helpers/userData.js#L30

**Comment by drteresavasquez** [comment, marked as answer]
That's probably not going to work the way you have it. What you need to do is compare all the pins data to the user data and do a merge using the spread and possibly a loop. Are these pins for one user or are you trying to do this for all the users?

**Reply by drteresavasquez** [reply]
If for multiple users, you're probably going to have to do something like this: https://github.com/drteresavasquez/IN-CLASS-almost-amazon-template/blob/main/src/javascripts/helpers/data/authorBooksData.js#L7

The thing is you're already getting all the pins so you just need to get the users whose pins they belong to. 

It's a complex thing but working through this will be good for you. Take your hands off the keyboard and THINK about what you're trying to do. Diagrams work 😊

**Reply by katherinevfry** [reply]
Thank you! I'm going to draw this out and take a crack at it again! I am trying to do multiple users!

**Reply by drteresavasquez** [reply]
Cool! You can definitely do it but getting a good plan together and doing it line by line is the best bet.

**Reply by katherinevfry** [reply]
So... I still haven't figured out how to filter and merge the data (will have to revisit this ASAP. I am determined to figure this out).

However, I did realize there was a much simpler way to accomplish what I wanted: When creating a new pin, add the current user's photo and name to the pin object, since you already have access to the user data. 

So while I do still need to learn how to filter and merge data, TIL that it's best to think through how to do something completely, lest you end up spending 7+ hours on something that could have taken 15 minutes if done a different way. 😬🤣

**Reply by drteresavasquez** [reply]
Love that you found a different way! Sometimes that's the best way to learn. Make it work first! 

The downside to this is if the user updates anything, it won't update on the pin or board BUT standing up the concept is the biggest part and I can buy that solution!

**Reply by drteresavasquez** [reply]
@katherinevfry go ahead and give us a rundown of your solution when you can so I can mark it as the answer.

**Comment by drteresavasquez** [comment]
Oh also for your filter, you need a taco otherwise it won't work @katherinevfry
